test/mocks/mockGroupCloudDataTopics.js

```diff
@@ -0,0 +1,51 @@
+var sinon = require('sinon');
+require('sinon-as-promised');
+
+/**
+ * This function builds a sinon stub to define the request() behaviour of fh-wfm-mediator's topic.request
+ * to be used by the group router
+ * @returns {stub}
+ */
+function getRequestStub() {
+  var requestStub = sinon.stub();
+  var mockGroupList = [
+    {id: "group-test-id-1", name: 'Drivers', role: 'worker'},
+    {id: "group-test-id-2", name: 'Back Office', role: 'manager'},
+    {id: "group-test-id-3", name: 'Management', role: 'admin'}
+  ];
+  var mockUpdatedGroup = {id: 'group-test-id-1', name: 'Drivers-UPDATED', role: 'worker'};
+  var mockCreatedGroup = {id: 'group-test-generated-id', name: 'New Group', role: 'worker'};
+
+  var mockIdSuc = 'group-test-id-1';
+  var mockIdNA = 'group-test-id-NA';
+  var mockIdErr = 'group-test-id-ERR';
+
+  //List
+  requestStub.withArgs('list').resolves(mockGroupList);
+
+  //Read
+  requestStub.withArgs('read', mockIdSuc).resolves(mockGroupList[0]);
+  requestStub.withArgs('read', mockIdNA).resolves({});
+  requestStub.withArgs('read', mockIdErr).rejects(new Error("An error occurred during group read"));
+
+  //Update
+  requestStub.withArgs('update', sinon.match.object, sinon.match({uid: mockIdSuc})).resolves(mockUpdatedGroup);
+  requestStub.withArgs('update', sinon.match.object, sinon.match({uid: mockIdNA})).rejects(new Error("Group to be updated does not exist"));
+  requestStub.withArgs('update', sinon.match.object, sinon.match({uid: mockIdErr})).rejects(new Error("An error occurred during group update"));
+
+  //Create
+  requestStub.withArgs('create', sinon.match.object, sinon.match.object).resolves(mockCreatedGroup);
+
+  //Delete
+  requestStub.withArgs('delete', sinon.match.object, sinon.match({uid: mockIdSuc})).resolves(mockGroupList[0]);
+  requestStub.withArgs('delete', sinon.match.object, sinon.match({uid: mockIdNA})).rejects(new Error("Group to be deleted does not exist"));
+  requestStub.withArgs('delete', sinon.match.object, sinon.match({uid: mockIdErr})).rejects(new Error("An error occurred during group deletion"));
+
+  requestStub.throws("Invalid Arguments");
+
+  return requestStub;
+}
```

Comment:
I think it would be best to setup an actual `mediator` that publishes fixture data instead of mocking at the `topics.request` level, setting it up for each test instead of using `.withArgs` to return different data increases locality of reference and independency between tests.
Having to look at multiple files to understand how a unit test works will make the unit tests themselves quite hard to maintain! and using more abstractions (such as superpowered sinon stubs) increases the mental load.
It would also serve as a way to see what kind of data we expect to see on each topic.

An example of what I mean is [here](https://github.com/feedhenry-raincatcher/raincatcher-mediator/blob/development/lib/topics/index-spec.js), it can be more verbose but changing one test will never break another, and each test is self-contained.